### PR TITLE
Iluise/head

### PR DESCRIPTION
**Pull request by iluise**
restore new settings to run with batch size 96.

### CODE REVIEW

File: atmorep/core/atmorep_model.py [hunk 1]
```diff
@@ -137,9 +137,11 @@ def mode( self, mode : NetMode) :
     
     if mode == NetMode.train :
       self.data_loader_iter = iter(self.data_loader_train)
+      #self.data_loader_iter = iter(self.dataset_train)
```
Comment (clessig) [inline atmorep/core/atmorep_model.py:140]: Can we remove this; same a few lines below.

File: atmorep/core/evaluate.py [hunk 2]
```diff
@@ -41,7 +41,7 @@
   #                  global_forecast_range
   # options can be used to over-write parameters in config; some modes also have specific options, 
   # e.g. global_forecast where a start date can be specified
-
+
```
Comment (clessig) [inline atmorep/core/evaluate.py:44]: Remove whitespace, just so that we have a clean diff.

File: atmorep/core/atmorep_model.py [hunk 3]
```diff
@@ -187,15 +189,15 @@ def create( self, pre_batch, devices, create_net = True, pre_batch_targets = Non
                                                 cf.batch_size,
                                                 pre_batch, cf.n_size, cf.num_samples_per_epoch,
                                           with_shuffle = (cf.BERT_strategy != 'global_forecast'), 
-                                                with_source_idxs = True )
+                                                with_source_idxs = True, compute_weights = (cf.losses.count('weighted_mse') > 0) )
```
Comment (clessig) [inline atmorep/core/atmorep_model.py:192]: This breaks the 100 character line limit.

File: atmorep/core/evaluator.py [hunk 4]
```diff
@@ -82,7 +77,7 @@ def evaluate( mode, model_id, file_path, args = {}, model_epoch=-2) :
     else :
       num_accs_per_task = int( 4 / int( os.environ.get('SLURM_TASKS_PER_NODE', '1')[0] ))
     devices = init_torch( num_accs_per_task)
-    #devices = ['cuda']
+    devices = ['cuda:1']
```
Comment (clessig) [inline atmorep/core/evaluator.py:80]: This should be commented out again, right?

File: atmorep/datasets/multifield_data_sampler.py [hunk 5]
```diff
@@ -217,29 +218,29 @@ def __iter__(self):
...
-            
-            grid = np.flip(np.array( np.meshgrid( lons_idx, lats_idx)), axis = 0) #flip to have lat on pos 0 and lon on pos 1
-            grid = torch.from_numpy( np.array( np.broadcast_to( grid,
-                                     shape = [tok_size[0]*num_tokens[0], *grid.shape])).swapaxes(0,1))
+              idx_base = tmidx_list[ifield][ilevel][ibatch]
+              idx_loc = idx_base - np.prod(num_tokens) * ibatch
+              
+              grid = np.flip(np.array( np.meshgrid( lons_idx, lats_idx)), axis = 0) #flip to have lat on pos 0 and lon on pos 1
+              grid = torch.from_numpy( np.array( np.broadcast_to( grid,
+                                      shape = [tok_size[0]*num_tokens[0], *grid.shape])).swapaxes(0,1))
```
Comment (clessig) [inline atmorep/datasets/multifield_data_sampler.py:235]: What do we do about the perf issues here? Leave it in and open an issue for it?

**Review by clessig — CHANGES_REQUESTED** [review]

File: atmorep/datasets/multifield_data_sampler.py (hunk 5 above)
Comment (iluise) [reply atmorep/datasets/multifield_data_sampler.py:235]: yup, I opened the issue: https://github.com/clessig/atmorep/issues/26

**Comment by iluise** [comment]
Hi, 

I should have solved all conflicts in the new commit.

**Review by clessig — APPROVED** [review]

**Pull request merged**